### PR TITLE
CMake compilation fixes

### PR DESCRIPTION
**Pull request by rbberger**
## Purpose
Fixes various issues when trying to compile using CMake
- add missing system header in MANYBODY files
- updated CMake build for USER-COLVARS
- remove dependency of `BUILD_SHARED_LIBS` when building PYTHON package (it is only necessary if you want to use the LAMMPS Python module, which isn't a hard requirement)

## Author(s)

Richard Berger (Temple U)

## Post Submission Checklist

- [x] The feature or features in this pull request is complete
- [x] The source code follows the LAMMPS formatting guidelines

### CODE REVIEW

File: src/MANYBODY/pair_polymorphic.h [hunk 1]
```diff
@@ -21,6 +21,7 @@ PairStyle(polymorphic,PairPolymorphic)
 #define LMP_PAIR_POLYMORPHIC_H
 
 #include "pair.h"
+#include <math.h>
```
Comment (junghans) [inline src/MANYBODY/pair_polymorphic.h]: Why did this only show up in CMake?

File: cmake/CMakeLists.txt [hunk 2]
```diff
@@ -401,6 +400,14 @@ foreach(SIMPLE_LIB REAX MEAM POEMS USER-ATC USER-AWPMD USER-COLVARS USER-H5MD
       target_include_directories(awpmd PUBLIC ${LAMMPS_LIB_SOURCE_DIR}/awpmd/systems/interact ${LAMMPS_LIB_SOURCE_DIR}/awpmd/ivutils/include)
     elseif(PKG_LIB STREQUAL h5md)
       target_include_directories(h5md PUBLIC ${LAMMPS_LIB_SOURCE_DIR}/h5md/include)
+    elseif(PKG_LIB STREQUAL colvars)
+      file(GLOB_RECURSE LEPTON_SOURCES ${LAMMPS_LIB_SOURCE_DIR}/colvars/lepton/src/*.cpp)
```
Comment (junghans) [inline cmake/CMakeLists.txt]: Why that double include the cpp fils in `colvars/lepton/src`as there is already a `file(GLOB_RECURSE ${PKG_LIB}_SOURCES ... ${LAMMPS_LIB_SOURCE_DIR}/${PKG_LIB}/*.cpp)`?

And do the Makesfiles build this a two libraries as well?

Maybe `USER-COLVARS` is better build outside of the `SIMPLE_LIB` loop.

File: src/MANYBODY/pair_polymorphic.h (hunk 1 above)
Comment (rbberger) [reply src/MANYBODY/pair_polymorphic.h]: ¯\_(ツ)_/¯

File: cmake/CMakeLists.txt (hunk 2 above)
Comment (rbberger) [reply cmake/CMakeLists.txt]: Simplified it.

File: cmake/CMakeLists.txt [hunk 3]
```diff
@@ -401,13 +401,8 @@ foreach(SIMPLE_LIB REAX MEAM POEMS USER-ATC USER-AWPMD USER-COLVARS USER-H5MD
...
       target_include_directories(h5md PUBLIC ${LAMMPS_LIB_SOURCE_DIR}/h5md/include)
     elseif(PKG_LIB STREQUAL colvars)
-      file(GLOB_RECURSE LEPTON_SOURCES ${LAMMPS_LIB_SOURCE_DIR}/colvars/lepton/src/*.cpp)
-      add_library(lepton STATIC ${LEPTON_SOURCES})
-      target_include_directories(lepton PUBLIC ${LAMMPS_LIB_SOURCE_DIR}/colvars/lepton/include)
-
       target_compile_options(colvars PRIVATE -DLEPTON)
-      target_link_libraries(colvars lepton)
-      target_include_directories(colvars PUBLIC ${LAMMPS_LIB_SOURCE_DIR}/colvars)
+      target_include_directories(colvars PUBLIC ${LAMMPS_LIB_SOURCE_DIR}/colvars/lepton/include ${LAMMPS_LIB_SOURCE_DIR}/colvars)
```
[the remark this answers is not in the document]
Comment (rbberger) [reply cmake/CMakeLists.txt]: @giacomofiorin ok, done.

**Pull request merged**